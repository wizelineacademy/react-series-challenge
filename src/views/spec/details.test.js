import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';

import { Details } from '../';

const mockStore = configureMockStore();

describe('<Details />', () => {
  const props = {
    fetchGif: jest.fn(),
    toggleFavorite: jest.fn(),
    match: { params: { id: 'asdfljlk3-ffgfg' } },
  };

  test('renders null if no data and not loading', () => {
    const initialState = {
      favorites: [],
      gif: {
        data: undefined,
        loading: false,
      },
    };
    const tree = renderer.create(
      <Provider store={mockStore(initialState)}>
        <StaticRouter location="" context={{}}>
          <Details {...props} />
        </StaticRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders loader', () => {
    const initialState = {
      favorites: [],
      gif: {
        data: undefined,
        loading: true,
      },
    };
    const tree = renderer.create(
      <Provider store={mockStore(initialState)}>
        <StaticRouter location="" context={{}}>
          <Details {...props} />
        </StaticRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders error message', () => {
    const initialState = {
      favorites: [],
      gif: {
        data: undefined,
        loading: true,
        error: { message: 'Something went wrong' },
      },
    };
    const tree = renderer.create(
      <Provider store={mockStore(initialState)}>
        <StaticRouter location="" context={{}}>
          <Details {...props} />
        </StaticRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders data correctly ', () => {
    const initialState = {
      favorites: [],
      gif: {
        data: {
          data: {
            id: 'asdfljlk3-ffgfg',
            title: 'Example image',
            images: {
              original: {
                url: 'http://www.example.com/image.jpg',
                width: '800',
                height: '900',
              },
            },
            user: {
              username: 'krilbert',
              display_name: 'Alfer',
              profile_url: 'http://www.example.com',
              avatar_url: 'http://www.example.com/avatar.jpg',
            },
          },
        },
        loading: false,
      },
    };
    const tree = renderer.create(
      <Provider store={mockStore(initialState)}>
        <StaticRouter location="" context={{}}>
          <Details {...props} />
        </StaticRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
