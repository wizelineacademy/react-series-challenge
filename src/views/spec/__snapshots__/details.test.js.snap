// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> renders data correctly  1`] = `
<div
  className="sc-jTzLTM bRoPow"
>
  <div
    className="sc-fjdhpX kbMDXR"
  >
    <div
      className="sc-dnqmqq cWEUzS"
    >
      <a
        href="http://www.example.com"
      >
        <img
          alt="Alfer"
          className="sc-htoDjs iOyEEs"
          src="http://www.example.com/avatar.jpg"
        />
      </a>
      <div>
        Alfer
      </div>
      <div>
        @krilbert
         
        
      </div>
    </div>
  </div>
  <div
    className="sc-cSHVUG ffcNiE"
  >
    <img
      alt="Example image"
      className="sc-kAzzGY bYEREJ"
      src="http://www.example.com/image.jpg"
    />
  </div>
  <div
    className="sc-jzJRlG NGNYY"
  >
    <button
      onClick={[Function]}
    >
      Add to Favorites
    </button>
  </div>
</div>
`;

exports[`<Details /> renders error message 1`] = `
<div>
  Something went wrong
</div>
`;

exports[`<Details /> renders loader 1`] = `
<div
  className="sc-jTzLTM bRoPow"
>
  <div
    className="sc-fjdhpX kbMDXR"
  >
    <div
      className="sc-bwzfXH kmXviU"
    >
      <div
        animation-delay={0}
        className="sc-htpNat gLLNR"
      />
      <div
        animation-delay="-1.1s"
        className="sc-htpNat gteQUn"
      />
      <div
        animation-delay="-1.0s"
        className="sc-htpNat dnZQtX"
      />
      <div
        animation-delay="-0.9s"
        className="sc-htpNat fqexcQ"
      />
      <div
        animation-delay="-0.8s"
        className="sc-htpNat gcoKLq"
      />
    </div>
  </div>
  <div
    className="sc-cSHVUG ffcNiE"
  >
    <div
      className="sc-bwzfXH kmXviU"
    >
      <div
        animation-delay={0}
        className="sc-htpNat gLLNR"
      />
      <div
        animation-delay="-1.1s"
        className="sc-htpNat gteQUn"
      />
      <div
        animation-delay="-1.0s"
        className="sc-htpNat dnZQtX"
      />
      <div
        animation-delay="-0.9s"
        className="sc-htpNat fqexcQ"
      />
      <div
        animation-delay="-0.8s"
        className="sc-htpNat gcoKLq"
      />
    </div>
  </div>
  <div
    className="sc-jzJRlG NGNYY"
  />
</div>
`;

exports[`<Details /> renders null if no data and not loading 1`] = `null`;
